### PR TITLE
[deploy] Add deploy script

### PR DESCRIPTION
**Pull request by mrcnski**
# PULL REQUEST

## Overview

Adds a deploy script.

### CODE REVIEW

**Comment by mrcnski** [comment]
Waiting until SkynetLabs/nodejs-skynet#35 is merged so I can remove the second client in this PR.

File: deploy.js
Comment (interstella-5555) [inline deploy.js]: if this is going to be our official copy & paste deploy script we should minimise the number of dependencies we add, namely we can safely remove chalk here

**Review by interstella-5555 — CHANGES_REQUESTED** [review]

File: deploy.js
Comment (kwypchlo) [inline deploy.js:96]: this looks sus

**Review by kwypchlo — CHANGES_REQUESTED** [review]

**Review by interstella-5555 — DISMISSED** [review]

File: deploy.js
Comment (interstella-5555) [inline deploy.js]:
```suggestion
  return await client.uploadDirectory(path, prepareUploadOptions());
```

File: deploy.js
Comment (interstella-5555) [inline deploy.js]:
```suggestion
  return await client.registry.getEntryLink(publicKey, dataKey);
```

**Review by interstella-5555 — COMMENTED** [review]
is there a reason why you declare unused variables ?

**Review by interstella-5555 — DISMISSED** [review]

File: deploy.js
Comment (ro-tex) [inline deploy.js:20]: What happens if you get tryfiles with spaces in their names?
I suggest we use JSONArray instead of a space-separated string.

Example: https://dev3.siasky.dev/skynet/metadata/AADHGZCBA3LnCASvMeAPZIgvZ-ynUx1Ge_YU5MJJdZqhMw

File: deploy.js
Comment (ro-tex) [inline deploy.js]: Why do we have this conditional "s"? Do we expect to not have the `resolverSkylinkUrl` in some case? If that's the case we should check for it when outputting it with a `console.log` later on (line 100). But it seems that if we fail to get/update the resolver link we'll throw an exception anyway.

File: deploy.js
Comment (interstella-5555) [reply deploy.js:20]: you need to urlencode the file names yourself, I think that's fair trade off for such simplification

99% of the time it will be just either omitted, `index.html` or `/index.html index.html` anyway

Comment (ro-tex) [reply deploy.js:20]: I am not sure this approach is more self-explanatory than a JSONArray. I'm also not sure it will be self-evident for the devs that they need to URL-encode their filenames. In your example `/index.html` is not URL-encoded. I am sure most devs won't think of encoding that.

If, on the other hand, they configure their `tryfiles` as a JSONArray we can do the encoding for them in the DAC.

Comment (interstella-5555) [reply deploy.js:20]: nginx does it the same way http://nginx.org/en/docs/http/ngx_http_core_module.html#try_files

**Review by ro-tex — APPROVED** [review]
I'm still not sure that choosing `TRY_FILES` to be a flat string is a good idea but if you like it that way and Karol supports it then I won't block it.

File: deploy.js
Comment (ro-tex) [reply deploy.js:20]: We should at least add some validation then - does the string need to be url-encoded, etc.

**Review by peterjan — APPROVED** [review]

**Pull request merged**
